doc/radosgw/cloud-transition.rst

```diff
@@ -0,0 +1,325 @@
+                                               --storage-class CLOUDTIER \
+                                               --tier-config-rm=acls.source_id=testid
+
+    # radosgw-admin zonegroup placement modify --rgw-zonegroup default \
+                                               --placement-id default-placement \
+                                               --storage-class CLOUDTIER \
+                                               --tier-config-rm=target_path
+
+The storage class can be removed using the following command
+
+::
+
+    # radosgw-admin zonegroup placement rm --rgw-zonegroup={zone-group-name} \
+                                           --placement-id={placement-id} \
+                                           --storage-class={storage-class-name}
+
+For example,
+
+::
+
+    # radosgw-admin zonegroup placement rm --rgw-zonegroup default \
+                                           --placement-id default-placement \
+                                           --storage-class CLOUDTIER
+    [
+        {
+            "key": "default-placement",
+            "val": {
+                "name": "default-placement",
+                "tags": [],
+                "storage_classes": [
+                    "STANDARD"
+                ]
+            }
+        }
+    ]
+
+Object modification & Limitations
+----------------------------------
+
+The cloud storage class once configured can then be used like any other storage class in the bucket lifecyle rules. For example,
+
+::
+
+    <Transition>
+      <StorageClass>CLOUDTIER</StorageClass>
+      ....
+      ....
+    </Transition>
+
+
+Since the transition is unidirectional, while configuring S3 lifecycle rules, the cloud storage class should be specified last among all the storage classes the object transitions to. Subsequent rules (if any) do not apply post transition to the cloud.
+
+To avoid object names collision across various buckets, source bucket name is prepended to the target object name. If the object is versioned, object versionid is appended to the end.
+
+Below is the sample object name format:
+::
+
+    s3://<target_path>/<source_bucket_name>/<source_object_name>(:<source_object_version_id>)
+
+Due to API limitations there is no way to preserve original object modification time and ETag but they get stored as metadata attributes on the destination objects, as shown below:
+
+::
+
+   x-amz-meta-rgwx-source: rgw
+   x-amz-meta-rgwx-source-etag: ed076287532e86365e841e92bfc50d8c
+   x-amz-meta-rgwx-source-key: lc.txt
+   x-amz-meta-rgwx-source-mtime: 1608546349.757100363
+   x-amz-meta-rgwx-versioned-epoch: 0
+
+By default, post transition, the source object gets deleted. But it is possible to retain its metadata but with updated values (like storage-class and object-size) by setting config option 'retain_object' to true. However GET on those objects shall still fail with 'InvalidObjectState' error.
```

Comment:
I think we agreed that redirect or read-through would most likely be follow-on projects?

Reply:
yes.. This comment was earlier to that.. I shall add the same in the doc.

Reply:
added it to future work in the doc.

Reply:
Two questions.  First, this still seems ambiguous to me?  Delete implies ordinary deletion, which would be deletion from the source bucket index as well as on disk.  Is that currently the case?  Second, Casey mentioned you asked about object lock during transition.  I would vote that except in special circumstances (and by this I mean, to be defined later, I'm not sure they're well defined yet), we prohibit transition when an object is under object lock (retention or otherwise).  What do you think?

Reply:
delete (default case) here refers to deletion from both bucket index and disk.

yes. Object lock/retention is not currently handled. I too vote that it may be better to prohibit transition to cloud (as special case) if object is locked. 
I have one doubt. In the code currently, object_lock_retention checks ('pass_object_lock_check()') are done only for "LCOpAction_NonCurrentExpiration" case, but not for "LCOpAction_CurrentExpiration". Is this not applicable to current objects?

Reply:
Done. As discussed over irc, if its current version object which is transitioned to cloud, a delete marker is created for it . And non-current version locked objects are skipped from transition. Updated the doc too. Kindly review.